### ISSUE TITLE
Why does trainer.test() with ddp launch new processes?

### ISSUE THREAD

**Issue by sneiman** [PyTorchLightning/pytorch-lightning#746]
When using ddp, I have noticed that pl relaunches all the process to run testing. This does burn up a lot of time - and makes it challenging to share values that would be useful in the final test_end() reporting/logging - elapsed time or comparative statistics, for example.

Just curious as to why the processes are relaunched instead of simply running them like validation, but with the test dataset?

thx,

s

**Comment by sneiman** [comment]
Additional issue: this may be a bug. trainer.test(model) is currently producing random results - as if the network were untrained. This net in other training modes has produced much better results. Perhaps not only are the processes being relaunched but the model has been re-initialized as well?

**Comment by williamFalcon** [comment]
why not  change the backend  to DP and then run test?
there are instances when your test set (imagenet or something big) needs to be distributed.

**Comment by Borda** [comment]
btw, is there is a connection to #745?

**Comment by sneiman** [comment]
I submitted pr on another aspect of this, and included a NOTE about this particular problem. I will run with dp and see what happens.

**Comment by sneiman** [comment]
I did some testing - putting all in issue 747 for conciseness.

**Comment by Borda** [comment]
ok, then I close this issue in favour to #747
